libapicore/ApiServer.cpp

```diff
@@ -19,9 +20,12 @@ void ApiServer::start()
     if (m_portnumber == 0)
         return;
 
+    if(m_address.empty())
```

Comment:
Try parsing the address to boost's endpoint before using it to catch errors of badly formatted addresses.

Reply:
this was just setting the default value. i should have set it in m_address declaration instead. in fact, i made no effort whatsoever to detect bad input.